### ISSUE TITLE
Nedmalloc crashes on start

### ISSUE THREAD

**Issue by wolfpld** [ned14/nedmalloc#11]
This is a followup to bug #10. When using a6a0decf6714804ce479a87dc768cd7d76d20ec3 the application crashes immediately after starting.

signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 00000008
Stack frame #00  pc 0000c838  /system/lib/libc.so (memset+112): Unable to open symbol file obj/local/armeabi-v7a//libc.so. Error (9): Bad file descriptor
Stack frame #01  pc 000cc4d8  /data/data/com.darkrl.darkrl/lib/libDarkRL.so: Routine init_user_mstate in jni/../../../contrib/nedmalloc/malloc.c.h:5383

gcc-4.7 armeabi-v7a debug

**Comment by ned14** [comment]
That implies that memory returned by the system is not accessible because init_user_mstate segfaults on first write. I can't see how this would happen: here is the sequence:

```
char* tbase = (char*)(CALL_MMAP(tsize, 0));
if (tbase != CMFAIL) {
  m = init_user_mstate(tbase, tsize);
```

I'll look into it when I get home from work, but it's hard to see what's causing your report here.

Niall

**Comment by ned14** [comment]
Sorry, twas a typo. Fixed in eec9258912ed566c03e3aa018986420d2f93cdc5.
